ios/macOS: Return advertised name

cadivus opened the pull request:
Hi,
first of all, thanks for your work! This library helped me a lot.

There is a "weird" behavior about MacOS and iOS devices:
On other platforms, you see the advertised name. E.g. if you give your Bose headset a name, you would see something like "cadivus Bose QC45" instead of "Bose QC45". With this Darwin implementation, you will always see "Bose QC45".  
This PR addresses that so that this lib acts the same on every platform.

There is one special thing about system devices: iOS doesn't cache the advertised names. That's why I added that too.

The following videos show the difference. I use my Thinkpad to mock a BLE device. If you see this name, it's not the advertised name.

Without changes:  

https://github.com/user-attachments/assets/07a6f94c-2bef-4589-9622-01a197184d1f


Without caching:  


https://github.com/user-attachments/assets/5d8c5496-20e6-4b61-9618-e01ba871e978


With the chance incl caching:


https://github.com/user-attachments/assets/bc3e4cf6-47df-4e46-9dbc-d3118ff26d81

cadivus commented on darwin/Classes/UniversalBlePlugin.swift:
```diff
@@ -337,14 +343,18 @@ private class BleCentralDarwin: NSObject, UniversalBlePlatformChannel, CBCentral
       manufacturerDataList.append(universalManufacturerData!)
     }
 
+    let advertisedName = advertisementData[CBAdvertisementDataLocalNameKey] as? String
+    let displayName = advertisedName ?? peripheral.name
+    advertisementNameCache[peripheral.uuid.uuidString] = displayName
```
I don't think that's right. At this position, the advertised name should be accessible. If it isn't, there may be none.

fotiDim reviewed (changes requested):
@cadivus thank you for the very detailed PR 🙏. Indeed I confirm that the advertized name now is correctly reflected on iOS/macOS when changed ✅.
However, I was not able to replicate the "iOS doesn't cache the advertised names" issue you described. Could you do the following things?
- Make the necessary changes in the example app to allow replicating the issue. In my case I commented out `if (isConnected) UniversalBle.disconnect(widget.deviceId);` and then in the app I got the system devices (after setting a service filter).
- Split the above changes and caching addition into separate commits or even better into a separate PR.

cadivus commented:
@fotiDim I split the commits and modified the example. I still have issues without caching.  
  
This is what the full change looks like:  

https://github.com/user-attachments/assets/df3b3514-fe60-4057-b7d5-09d5be7987ca

<br>

This is what it looks like if I drop the cache commit (for system devices, the device name is not the advertised one:  

https://github.com/user-attachments/assets/2f21eb65-9954-4dde-ab6e-0f744c720d62

cadivus commented:
> No matter what I tried I couldn't reproduce. I reversed the caching commit, and then tried the exact same steps as in your macOS video

Did you completely rebuild it, not just do a hot reload?



> Are you certain that Android and the other platforms return the advertised name when getting system devices?

This is https://github.com/Navideck/universal_ble/pull/162 with the cherry picked example change (so no mods to the lib):

https://github.com/user-attachments/assets/19be4736-1256-4bec-8038-118719abff96

So yes, I am.

cadivus commented:
> and merge the PR as is.

Could you please create a new release soon. If this has to be made by hand all the time, maybe this PR helps you:
https://github.com/Navideck/universal_ble/pull/164
